src/Microsoft.TestPlatform.CommunicationUtilities/Interfaces/Communication/ICommunicationChannel.cs

```diff
@@ -24,5 +27,66 @@ public interface ICommunicationChannel : IDisposable
     /// Notification from server/client that data is available.
     /// </summary>
     /// <returns>A <see cref="Task"/> implying async nature of the function.</returns>
-    Task NotifyDataAvailable();
+    Task NotifyDataAvailable(CancellationToken cancellationToken);
+}
+
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
+public class TrackableEvent<T>
+#pragma warning restore CA1001 // Types that own disposable fields should be disposable
+{
+    private readonly ManualResetEventSlim _slim;
+
+    internal event EventHandler<T>? Event;
+
+    public TrackableEvent()
+    {
+        _slim = new ManualResetEventSlim(Event != null);
+    }
+
+    public void Notify(object sender, T eventArgs, string traceDisplayName)
+    {
+        var e = Event;
+        if (e != null)
+        {
+            e.SafeInvoke(sender, eventArgs!, traceDisplayName);
+        }
+    }
+
+    public bool WaitForSubscriber(int timeoutMilliseconds, CancellationToken cancellationToken)
+    {
+        var _ = timeoutMilliseconds;
+        var __ = cancellationToken;
+        return Event != null;
+        //return _slim.Wait(timeoutMilliseconds, cancellationToken);
```

Comment:
Reverting our test code, which we used to cause the original issue.
```suggestion
        return _slim.Wait(timeoutMilliseconds, cancellationToken);
```